src/typechecker.erl

```diff
@@ -1014,7 +1014,10 @@ type_check_expr_in(Env, ResTy, {tuple, LINE, TS}) ->
 	    % tuple in a union. That's potentially inefficient.
 	    % Maybe we should have a flag to allow for approximation here
 	    % in the same way that dialyzer does it.
-	    {VBs, Cs2} = type_check_tuple_in(Env, ResTy, TS),
+
+        %% normalize ResTy to resolve user and remote types
+        NormResTy = normalize(ResTy, Env#env.tenv),
```

Comment:
I think normalizing should be done for several clauses of `type_check_expr_in/4`, at least for cons, bin, tuple and record. I noticed this when I implemented `{bin, ...}`. Perhaps we can handle this in a single place, by adding a clause of type_check_expr_in for user and remote types, normalize and then a recursive call? Similar reasoning here as for normalizing in add_type_pat.